scipy/spatial/transform/rotation.py

```diff
@@ -281,3 +281,115 @@ def as_rotvec(self):
             return rotvec[0]
         else:
             return rotvec
+
+    @staticmethod
+    def _make_dcm_from_angle(char, angle, intrinsic=False):
+        sinx = np.sin(angle)
+        cosx = np.cos(angle)
+
+        axis = char.lower()
+        if axis == 'z':
+            dcm = np.array([
+                [cosx, -sinx, 0],
+                [sinx, cosx, 0],
+                [0, 0, 1]
+                ])
+        elif axis == 'y':
+            dcm = np.array([
+                [cosx, 0, sinx],
+                [0, 1, 0],
+                [-sinx, 0, cosx]
+                ])
+        elif axis == 'x':
+            dcm = np.array([
+                [1, 0, 0],
+                [0, cosx, -sinx],
+                [0, sinx, cosx]
+                ])
+
+        return dcm.transpose() if intrinsic else dcm
+
+    @classmethod
+    def _make_dcm_from_euler(cls, axis_str, angle):
+        extrinsic = recompile("^[xyz]{1,3}$")
+        intrinsic = recompile("^[XYZ]{1,3}$")
+
+        if len(axis_str) != angle.shape[0]:
```

Comment:
It seems to me that you don't check for invalid sequences. I can't claim that we agreed on that 100%, but it looks logical as we decided to stick to classic Euler angles. In this case we should allow sequence where the second axis is not equal to the first and the third axes.